lib/remote_node_manager.go

```diff
@@ -51,12 +51,19 @@ func NewRemoteNodeManager(srv *Server, bc *Blockchain, cmgr *ConnectionManager,
 	}
 }
 
-func (manager *RemoteNodeManager) newRemoteNode(validatorPublicKey *bls.PublicKey) *RemoteNode {
+func (manager *RemoteNodeManager) DisconnectAll() {
+	for _, rn := range manager.GetAllRemoteNodes().GetAll() {
```

Comment:
Is this thread-safe? What happens if a new remote node is added after `GetAllRemoteNodes()` returns

Reply:
I guess technically it is possible that some new connection came through as we're stopping the Node. But right now NetworkManager is stopped at the very end of Server's stop, after ConnectionManager, meaning as we get to NetworkManager.Stop, the ConnectionManager already seized to serve new incoming connections. Which makes the `DisconnectAll` implicitly thread-safe. If you're wondering if we should mutex-gate this method, I tried it initially, but it actually causes us to deadlock, since Disconnect also calls on the `mtx` mutex. That said, I think it's fine to leave this method without the mutex, as it's only called by the NetworkManager on exit. I will document this method a little more in a future PR.